### ISSUE TITLE
Dataset

### ISSUE THREAD

**Issue by ademenet** [Nielsyang/YOLOv1_tensorflow#1]
Hello,

I'm really interesting in your YOLO implementation. I found your work really smart.

But one question remains: how is your dataset?

I was looking to your `loss_func_yolo` in `very_tiny_yolo.py` (I'm trying to implement it right now), but I can't figure out how your datas are distributed. For example - line 212/213/214:

```
res += COORD_W * tf.sign(exp[i][j+2]) * (tf.square(output[i][j] - exp[i][j]) + tf.square(output[i][j+1]-exp[i][j+1]) + 
                                                                    tf.square(tf.sqrt(tf.abs(output[i][j+2])) - tf.sqrt(exp[i][j+2])) + 
                                                                    tf.square(tf.sqrt(tf.abs(output[i][j+3])) - tf.sqrt(exp[i][j+3])))
```

I guess `exp` stands for _expected_ and `output`, well, it talks by itself. Thus, for `tf.sign(exp[i][j+2])`: is that _objectness_? I conclude it should be 0 or 1. But why, a few lines later, you are using `tf.sign(tf.floor(exp[i][j]))`?

I think if you could share how looks `exp` and `output` it could be really useful.

Thank you very much :)

**Comment by lesliejackson** [comment]
Thanks for your advice.

The exp denotes expected output and output denotes model output in loss function,
here `tf.sign()` return 0 when there is no object in this grid,
return 1 when there is some objects in this grid.

Actually, `tf.sign()` is a characteristic function,  
you can see the loss function in https://arxiv.org/pdf/1506.02640.pdf.

Here I use `tf.sign(tf.floor())` because I don't know how to implement the characteristic function
when there is no object it returns 1 and there is some objects it returns 0,
so I let one of the coordinates(x,y,w,h) be equal to 1.00001 in the grid with no object.

Now `tf.sign(tf.floor(1.00001))` return 1 when there is no object in the grid
and `tf.sign(tf.floor(x))` return 0 when there is some objects in the grid
here coordinates(x,y,w,h) is normalized to (0,1).

Code is here
```  for i in range(labels.shape[0]):
    for j in range(labels.shape[1]):
      if j%(B*5+CLASSES) == 0 or j%(B*5+CLASSES) == 5:
        labels[i][j] = 1.00001
      else:
        labels[i][j] = 0

**Comment by ademenet** [comment]
Thank you very much for your explanations.

I see how you use `tf.sign()` now. Still, I'm wondering how do your `output` and `exp` look like?

In my case, my output is a `(None, 637)` vector because I have `7 rows * 7 cols * (2 bounding boxes * (4 coordinates + 1 objectness) + 3 classes)`. I reshape my ground truth so as to match this output. How did you do?

Moreover I use `objectness` in my ground truth (which is `0.` or `1.`) in `tf.sign()` rather than using the function you used (maybe I'm missing some details and maths background because I can't figure out why it is a characteristic function).

**Comment by ck196** [comment]
What kind of dataset do you use?
What is in `/home/yy/yolo_/train` and `/home/yy/yolo_/label`?

**Comment by lesliejackson** [comment]
@ck196 ，it's my own dataset just like Pascal Voc, detail at [Pascal Voc](http://host.robots.ox.ac.uk/pascal/VOC/)
